core/src/main/java/feign/Request.java

```diff
@@ -177,7 +177,9 @@ public String url() {
    * @return the current character set for the request, may be {@literal null} for binary data.
    */
   public Charset charset() {
-    return body.encoding;
+    return (body == null)
```

Comment:
This change will break the documentation, which states that this can return `null`